security/tls.go

```diff
@@ -115,18 +117,38 @@ func LoadInsecureTLSConfig() *tls.Config {
 }
 
 // LoadClientTLSConfigFromDir creates a client TLSConfig by loading the root CA certs from the
-// specified directory. The directory must contain ca.crt.
+// specified directory. The directory must contain the following files:
+// - ca.crt   -- the certificate of the cluster CA
+// - node.crt -- the certificate of this node; should be signed by the CA
+// - node.key -- the private key of this node
+// If the path is prefixed with "embedded=", load the embedded certs.
 func LoadClientTLSConfigFromDir(certDir string) (*tls.Config, error) {
+	certPEM, err := readFileFn(path.Join(certDir, "node.crt"))
+	if err != nil {
+		return nil, err
+	}
+	keyPEM, err := readFileFn(path.Join(certDir, "node.key"))
+	if err != nil {
+		return nil, err
+	}
 	caPEM, err := readFileFn(path.Join(certDir, "ca.crt"))
 	if err != nil {
 		return nil, err
 	}
-	return LoadClientTLSConfig(caPEM)
+
+	return LoadClientTLSConfig(certPEM, keyPEM, caPEM)
 }
 
-// LoadClientTLSConfig creates a client TLSConfig from the supplied byte strings containing
-// the certificate of the cluster CA.
-func LoadClientTLSConfig(caPEM []byte) (*tls.Config, error) {
+// LoadClientTLSConfig creates a client TLSConfig from the supplied byte strings containing:
+// - the certificate of this client (should be signed by the CA),
+// - the private key of this client.
```

Comment:
is "this client" going to be the local node (as a client to other nodes/itself)?